sympy/assumptions/assume.py

```diff
@@ -176,6 +176,10 @@ def eval(self, expr, assumptions=True):
                 except AttributeError:
                     continue
                 res = eval(expr, assumptions)
+                # Do not stop if Expr method returned None
+                # Try to check for higher classes
```

Comment:
The `Expr` method prevented the mro from evaluating `Basic`. I don't understand why it should be applied to all.